internal/pkg/client/ocisif/ocisif.go

```diff
@@ -282,5 +284,30 @@ func PushOCISIF(_ context.Context, sourceFile, destRef string, ociAuth *ocitypes
 		return fmt.Errorf("while obtaining image: %w", err)
 	}
 
-	return remote.Write(ir, image, ociauth.AuthOptn(ociAuth, reqAuthFile), remote.WithUserAgent(useragent.Value()))
+	remoteOpts := []remote.Option{ociauth.AuthOptn(ociAuth, reqAuthFile), remote.WithUserAgent(useragent.Value())}
+	if term.IsTerminal(2) {
+		pb := &progress.DownloadBar{}
+		progChan := make(chan ggcr.Update, 1)
+		go func() {
+			var total int64
+			soFar := int64(0)
+			for {
+				// The following is concurrency-safe because this is the only
+				// goroutine that's going to be reading progChan updates.
+				update := <-progChan
+				if update.Total != total {
+					pb.Init(update.Total)
+					total = update.Total
+				}
+				pb.IncrBy(int(update.Complete - soFar))
+				soFar = update.Complete
+				if soFar >= total {
+					return
```

Comment:
Should probably call `pb.Wait()` at this point, as that's the route to shutdown the progress bar / container go routines cleanly.

Note that Wait and Abort aren't seen called in the library flows as they called inside scs-library-client code.